accepted/2020/minimum-os-version/minimum-os-version.md

```diff
@@ -0,0 +1,245 @@
+
+If and when C# support target typing for enums, we could add convenience overloads that would improve readability:
+
+```csharp
+[Introduced (.macOS, 10, 1)]
+[Obsoleted (.macOS, 10, 8)]
+[Removed (.macOS, 10, 12)]
+```
+
+> NOTE: These attributes would only be needed on reference assemblies. They would not be needed at runtime. They could be removed by the linker, and we could also strip them from the implementation assemblies that we ship ourselves.
+
+# Runtime Checks
+
+> NOTE: See [this issue](https://github.com/dotnet/runtime/issues/33331) for a more active/current discussion of runtime check APIs.
+
+The following method will be added to the `RuntimeInformation` class in the `System.Runtime.InteropServices` namespace:
+
+```csharp
+public static bool CheckOSVersion(
+    int majorVersion, int minorVersion = 0, int servicingVersion = 0
+);
+```
+
+It is used as follows:
+
+```csharp
+if (RuntimeInformation.CheckOSVersion(15)) {
+    // use 15.0+ APIs
+} else {
+    // gracefully degrade
+}
+```
+
+There will be an additional overload that checks the OS as well as the OS version, for use in portable libraries:
+
+```csharp
+public static bool RuntimeInformation.CheckOS(
+    string platformIdentifier,
+    int majorVersion, int minorVersion = 0, int servicingVersion = 0
+);
+```
+
+As with the availability attributes, there could be an additional overload that used enums and target typing to improve readability.
+
+> NOTE: The linker, JIT and/or AOT compiler could treat these method as intrinsics with constant values, which would enable it to eliminate unnecessary fallback code when consuming portable libraries or libraries with a lower OS minimum version than the app itself.
+
+# MSBuild Properties
+
+The `Microsoft.Net.Sdk` targets will extract the OS API Version component from the `TargetFramework` into the `TargetPlatformVersion` MSBuild property and burn it into an assembly attribute. Tools that use this value are expected to access it from the MSBuild property or from the assembly attribute.
+
+Project files may specify a `MinimumPlatformVersion`, and if they do not it will default to the `TargetPlatformVersion` value. The `MinimumPlatformVersion` must not be higher than the `TargetPlatformVersion`, and this will be validated at the start of the build.
+
+It is recommended that platforms define more specifically named versions of this property of the form `Minimum{PlatformIdentifier}Version`, for example `MinimumIOSVersion` or `MinimumWindowsVersion`, and that their targets assign this value to `MinimumOSVersion` if it is set. This will simplify use of the property multi-targeting.
+
+> NOTE: would `{PlatformIdentifier}MinimumVersion` be better? It would sort/group more easily with other OS-specific properties.
+
+> NOTE: Using the existing `TargetPlatformVersion` property to represent the API version may be confusing for platforms such as Android that allow having different values for the target version and API version..
```

Comment:
when would you want to specify a different target and API version?

Reply:
Android always target the API levels. So `*MinVersion` -> becomes minimum API level, `*MaxVersion` becomes targeted API level and we can have `TargetPlatformVersion` point to either of these depending on the project or have new value that mirrors the behaviour of `TargetPlatformMaxVersionTested`.

Before you guys ask, yes, Windows SDK has more than 4 properties just for versions and 3 just for UWP project files!